### ISSUE TITLE
Result of one-phase local aggregation is wrong

### ISSUE THREAD

**Issue by ZiheLiu** [StarRocks/starrocks#31869]
### Steps to reproduce the behavior (Required)
3 BEs.

```sql
CREATE TABLE `t2` (
  `k0` int(11) NULL COMMENT "",
  `k1` int(11) NOT NULL COMMENT "",
  `k2` varchar(10) NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`k0`, k1, k2)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`k0`, k1, k2) BUCKETS 48
PROPERTIES (
"replication_num" = "1"
);
insert into t2 select null, 390, 'qb' from TABLE(generate_series(1, 10));
insert into t2 select null, 390, 'qb' from TABLE(generate_series(1, 10));

set enable_per_bucket_optmize = true;
select k0, k1, k2, count(1) cnt from t2 group by k0, k1, k2;

```
### Expected behavior (Required)
```
+--------+-----+----+-----+
| k0     | k1  | k2 | cnt |
+--------+-----+----+-----+
| <null> | 390 | qb | 20  |
+--------+-----+----+-----+
```

### Real behavior (Required)
```

+--------+-----+----+-----+
| k0     | k1  | k2 | cnt |
+--------+-----+----+-----+
| <null> | 390 | qb | 10  |
| <null> | 390 | qb | 10  |
+--------+-----+----+-----+
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
80b0a6c

**Comment by ZiheLiu** [comment]
Fixed by #31837.

**Comment by ZiheLiu** [comment]
Fixed by #31837
